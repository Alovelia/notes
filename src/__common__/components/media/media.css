.Media {
  display: flex;
  align-items: flex-start;
}

.Media__figure {
  margin-right: 1rem;
}
  .Media__figure--right{
    margin-right: 0;
    margin-left: 1rem;
  }

.Media__body {
  flex: 1;
}
