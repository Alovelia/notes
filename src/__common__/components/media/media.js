import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './media.css';

function renderChild(node, i) {
  if (node.type.name === 'MediaFigure' && i !== 0) {
    const className = cx(node.props.className, 'Media__figure--right');
    return React.cloneElement(node, { className });
  }
  return node;
}

export const Media = props => (
  <div className="Media">
    {React.Children.map(props.children, renderChild)}
  </div>);

Media.propTypes = {
  children: PropTypes.node.isRequired
};

export const MediaFigure = props => (
  <div className={cx('Media__figure', props.className)}>
    {props.children}
  </div>);

MediaFigure.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

MediaFigure.defaultProps = {
  className: null
};

export const MediaBody = props => (
  <div className="Media__body">
    {props.children}
  </div>);

MediaBody.propTypes = {
  children: PropTypes.node.isRequired
};

