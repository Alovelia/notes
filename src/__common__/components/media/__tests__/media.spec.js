import { shallow } from 'enzyme';
import React from 'react';
import noop from 'lodash/noop';
import { Media, MediaBody, MediaFigure } from '../media';

describe('<Media />', () => {
  let wrapper;
  it('when initializing, the expected output is rendered', () => {
    const props = {
      // trigger: noop
    };
    const mockBody = <div>MOCK BODY</div>;
    const mockFigure = <div>MOCK FIGURE</div>;
    wrapper = shallow(
      <Media {...props}>
        <MediaFigure>{mockFigure}</MediaFigure>
        <MediaBody>{mockBody}</MediaBody>
      </Media>);
    expect(wrapper).toMatchSnapshot();
  });

  it('when initializing, the expected output is rendered if MediaFigure is not the first', () => {
    const props = {
      // trigger: noop
    };
    const mockBody = <div>MOCK BODY</div>;
    const mockFigure = <div>MOCK FIGURE</div>;
    wrapper = shallow(
      <Media {...props}>
        <MediaBody>{mockBody}</MediaBody>
        <MediaFigure>{mockFigure}</MediaFigure>
      </Media>);
    expect(wrapper).toMatchSnapshot();
  });
});
