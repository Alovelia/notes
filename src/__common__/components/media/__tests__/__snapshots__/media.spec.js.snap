// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Media /> when initializing, the expected output is rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Media>
    <MediaFigure
      className={null}
    >
      <div>
        MOCK FIGURE
      </div>
    </MediaFigure>
    <MediaBody>
      <div>
        MOCK BODY
      </div>
    </MediaBody>
  </Media>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MediaFigure
          className={null}
        >
          <div>
            MOCK FIGURE
          </div>
        </MediaFigure>,
        <MediaBody>
          <div>
            MOCK BODY
          </div>
        </MediaBody>,
      ],
      "className": "Media",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "function",
        "props": Object {
          "children": <div>
            MOCK FIGURE
          </div>,
          "className": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "MOCK FIGURE",
          },
          "ref": null,
          "rendered": "MOCK FIGURE",
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": ".1",
        "nodeType": "function",
        "props": Object {
          "children": <div>
            MOCK BODY
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "MOCK BODY",
          },
          "ref": null,
          "rendered": "MOCK BODY",
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MediaFigure
            className={null}
          >
            <div>
              MOCK FIGURE
            </div>
          </MediaFigure>,
          <MediaBody>
            <div>
              MOCK BODY
            </div>
          </MediaBody>,
        ],
        "className": "Media",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "children": <div>
              MOCK FIGURE
            </div>,
            "className": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "MOCK FIGURE",
            },
            "ref": null,
            "rendered": "MOCK FIGURE",
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "children": <div>
              MOCK BODY
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "MOCK BODY",
            },
            "ref": null,
            "rendered": "MOCK BODY",
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Media /> when initializing, the expected output is rendered if MediaFigure is not the first 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Media>
    <MediaBody>
      <div>
        MOCK BODY
      </div>
    </MediaBody>
    <MediaFigure
      className={null}
    >
      <div>
        MOCK FIGURE
      </div>
    </MediaFigure>
  </Media>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MediaBody>
          <div>
            MOCK BODY
          </div>
        </MediaBody>,
        <MediaFigure
          className="Media__figure--right"
        >
          <div>
            MOCK FIGURE
          </div>
        </MediaFigure>,
      ],
      "className": "Media",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "function",
        "props": Object {
          "children": <div>
            MOCK BODY
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "MOCK BODY",
          },
          "ref": null,
          "rendered": "MOCK BODY",
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": ".1",
        "nodeType": "function",
        "props": Object {
          "children": <div>
            MOCK FIGURE
          </div>,
          "className": "Media__figure--right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "MOCK FIGURE",
          },
          "ref": null,
          "rendered": "MOCK FIGURE",
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MediaBody>
            <div>
              MOCK BODY
            </div>
          </MediaBody>,
          <MediaFigure
            className="Media__figure--right"
          >
            <div>
              MOCK FIGURE
            </div>
          </MediaFigure>,
        ],
        "className": "Media",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "children": <div>
              MOCK BODY
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "MOCK BODY",
            },
            "ref": null,
            "rendered": "MOCK BODY",
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "children": <div>
              MOCK FIGURE
            </div>,
            "className": "Media__figure--right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "MOCK FIGURE",
            },
            "ref": null,
            "rendered": "MOCK FIGURE",
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
